cranelift/codegen/src/legalizer/heap.rs

```diff
@@ -198,5 +220,11 @@ fn compute_addr(
         pos.ins().global_value(addr_ty, base_gv)
     };
 
-    pos.func.dfg.replace(inst).iadd(base, offset);
+    if let Some(c) = spectre_oob_cond {
+        let final_addr = pos.ins().iadd(base, offset);
+        let zero = pos.ins().iconst(addr_ty, 0);
+        pos.func.dfg.replace(inst).select(c, zero, final_addr);
```

Comment:
Generating IR for the Spectre check seems quite dangerous, at this point; any optimization could reorganize the code (move it around in the best case, delete it in the worst case), so I don't think this is the right solution here.

Reply:
Even worse, there is a legalization for `select` that uses branches when there is no encoding that uses `cmov`:

https://github.com/bytecodealliance/wasmtime/blob/c9a3f05afd45961b0b397f97c4ad79cd7a7c807d/cranelift/codegen/src/legalizer/mod.rs#L476-L513

Reply:
Changed to use a specific, effectful instruction that the optimizer and legalizations do not know about.